src/ConnectionRequest.jl

```diff
@@ -34,13 +34,13 @@ end
 
 function getproxy(scheme, host)
     isnoproxy(host) && return nothing
-    if scheme == "http" && haskey(ENV, "http_proxy")
+    if scheme == "http" && !isempty(get(ENV, "http_proxy", ""))
         return ENV["http_proxy"]
```

Comment:
If you want to avoid two dict-lookups:
```suggestion
    if scheme == "http" && (p = get(ENV, "http_proxy", ""); !isempty(p))
        return p
```